Assets/Project/Scripts/Common/Entities/TreeId.cs

```diff
@@ -8,18 +9,26 @@ namespace Treevel.Common.Entities
     /// </summary>
     public enum ETreeId {
         Dummy = 0,  // ダミー
-        Spring_1 = 1,   // 春の木
-        Spring_2 = 2,
-        Spring_3 = 3,
-        Summer_1 = 1001,    // 夏の木
-        Summer_2 = 1002,
-        Summer_3 = 1003,
-        Autumn_1 = 2001,     // 秋の木
-        Autumn_2 = 2002,
-        Autumn_3 = 2003,
-        Winter_1 = 3001,    // 冬の木
-        Winter_2 = 3002,
-        Winter_3 = 3003,
+        Spring_1 = ESeasonId.Spring * Constants.MAX_TREE_NUM_IN_SEASON + 1,   // 春の木
+        Spring_2 = ESeasonId.Spring * Constants.MAX_TREE_NUM_IN_SEASON + 2,
+        Spring_3 = ESeasonId.Spring * Constants.MAX_TREE_NUM_IN_SEASON + 3,
+        Summer_1 = ESeasonId.Summer * Constants.MAX_TREE_NUM_IN_SEASON + 1,    // 夏の木
+        Summer_2 = ESeasonId.Summer * Constants.MAX_TREE_NUM_IN_SEASON + 2,
+        Summer_3 = ESeasonId.Summer * Constants.MAX_TREE_NUM_IN_SEASON + 3,
+        Autumn_1 = ESeasonId.Autumn * Constants.MAX_TREE_NUM_IN_SEASON + 1,     // 秋の木
+        Autumn_2 = ESeasonId.Autumn * Constants.MAX_TREE_NUM_IN_SEASON + 2,
+        Autumn_3 = ESeasonId.Autumn * Constants.MAX_TREE_NUM_IN_SEASON + 3,
+        Winter_1 = ESeasonId.Winter * Constants.MAX_TREE_NUM_IN_SEASON + 1,    // 冬の木
+        Winter_2 = ESeasonId.Winter * Constants.MAX_TREE_NUM_IN_SEASON + 2,
+        Winter_3 = ESeasonId.Winter * Constants.MAX_TREE_NUM_IN_SEASON + 3,
+    }
+
+    public static class ETreeIdExtension
+    {
+        public static ESeasonId GetSeasonId(this ETreeId treeId)
+        {
+            return (ESeasonId)((int)treeId / Constants.MAX_TREE_NUM_IN_SEASON);
```

Comment:
行数的には長いですが， `switch 文` を使った方が，コードとしては読みやすいと思いました

```suggestion
            switch (treeId) {
                case Spring_1:
                case Spring_2:
                case Spring_3:
                    return ESeasonId.Spring
                ...
            }
```

Reply:
`treeId`から`seasonId`を復元できるアルゴリズムにしてあるのに条件分岐で戻すのはなあ、、、という気持ちがある。
どっちでも良いっちゃ良いので @takesi0627 の意見待ちかなあ。

Reply:
了解！

Reply:
木が増える場合とか対応が増えるので、現状がいいと思います。